runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -212,6 +220,32 @@ public ClientUpdateCheckMode convert(String v)
 			}
 		});
 
+		SwingUtilities.invokeAndWait(() ->
+		{
+			RuneLiteSplashScreen.setTheme();
+			if (options.has("no-splash"))
+			{
+				SPLASH_SCREEN = new RuneLiteSplashScreen(new File(LOGS_DIR, "client.log"));
+			}
+			else
+			{
+				SPLASH_SCREEN = new RuneLiteSplashScreen("Runelite " + PROPERTIES.getVersion(), new File(LOGS_DIR, "client.log"));
+			}
+		});
+
+		SPLASH_SCREEN.setMessage("Checking launcher version", 0);
+		final String launcherVersion = RuneLiteProperties.getLauncherVersion();
+		if (launcherVersion != null &&
+			!EXPECTED_LAUNCHER_VERSION.equalsIgnoreCase(launcherVersion))
+		{
+			SwingUtilities.invokeAndWait(SPLASH_SCREEN::invalidVersion);
```

Comment:
Why is invalidVersion part of SPLASH_SCREEN? will it be used also in launcher? if no, why is it there

Reply:
I created the version check for the launcher but in order to force people to download the updated launcher we needed a version check in the client (at least temporarily)